### PR TITLE
Order by multiple properties sequentially

### PR DESCRIPTION
**Pull request by Serg046**
## IMPORTANT 

* [x] The code complies with the [Coding Guidelines for C#](https://www.csharpcodingguidelines.com/).
* [x] The changes are covered by a new or existing set of unit tests which follow the Arrange-Act-Assert syntax such as is used [in this example](https://github.com/fluentassertions/fluentassertions/blob/daaf35b9b59b622c96d0c034e8972a020b2bee55/Tests/FluentAssertions.Shared.Specs/BasicEquivalencySpecs.cs#L33).
* [x] If the contribution adds a feature or fixes a bug, please update [**the release notes**](https://github.com/fluentassertions/fluentassertions/tree/master/docs), which are published on the [website](https://fluentassertions.com/releases).
* [x] If the contribution changes the public API the changes needs to be included by running `AcceptApiChanges.ps1`/`AcceptApiChanges.sh`.
* [ ] If the contribution affects [the documentation](https://github.com/fluentassertions/fluentassertions/tree/develop/docs/_pages), please include your changes in this pull request so the documentation will appear on the [website](https://www.fluentassertions.com/introduction).

Issue #1160

### CODE REVIEW

**Comment by Serg046** [comment]
ThenNotBeInAscendingOrder and ThenNotBeInDescendingOrder are not provided because it will significantly complicate the change. The issue is the following lines of `NotBeOrderedBy`:
```csharp
Execute.Assertion
    .ForCondition(!unordered.SequenceEqual(expectation))
    .BecauseOf(because, becauseArgs)
    .FailWith("Expected {context:collection} {0} to not be ordered {1}{reason} and not result in {2}.",
        Subject, orderString, expectation);
```
It would be much more complex that just `!unordered.SequenceEqual(expectation)`.
Anyway, I don't think it is worth implementing because even if they were implemented, it would be difficult to understand their behavior therefore they would be in rare use.

File: Src/FluentAssertions/Collections/SubsequentOrderingGenericCollectionAssertions.cs [hunk 1]
```diff
@@ -0,0 +1,174 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Linq.Expressions;
+
+namespace FluentAssertions.Collections
+{
+    [DebuggerNonUserCode]
+    public class SubsequentOrderingGenericCollectionAssertions<T> :
```
Comment (dennisdoomen) [inline Src/FluentAssertions/Collections/SubsequentOrderingGenericCollectionAssertions.cs:10]: 🔧 Maybe rename it to `SubsequentOrderingAssertions`?

File: docs/_pages/releases.md [hunk 2]
```diff
@@ -16,6 +16,7 @@ sidebar:
 * Added `ComparingBy{Members,Value}(Type)` to allow specifying open generic types - [#1389](https://github.com/fluentassertions/fluentassertions/pull/1389).
 * Added overload of `CollectionAssertions.NotBeEquivalentTo` that takes a `config` parameter` - [#1408](https://github.com/fluentassertions/fluentassertions/pull/1408).
 * Changed `StringAssertions.StartWith`, `StringAssertions.EndWith` and their `EquivalentOf` versions to allow empty strings - [#1413](https://github.com/fluentassertions/fluentassertions/pull/1413).
+* Added ThenBeInAscendingOrder and ThenBeInDescendingOrder to allow asserting that a subsequence is ordered in ascending or descending order - [#1416](https://github.com/fluentassertions/fluentassertions/pull/1416).
```
Comment (dennisdoomen) [inline docs/_pages/releases.md:19]: 🙃 Would be nice to format those names using back-ticks.

File: Tests/FluentAssertions.Specs/Collections/GenericCollectionAssertionsSpecs.cs [hunk 3]
```diff
@@ -1284,6 +1284,118 @@ public void When_asserting_the_items_in_ay_collection_are_not_ordered_using_an_i
 
         #endregion
 
+        #region Then be in order
+
+        [Fact]
+        public void When_asserting_subsequence_by_then_be_in_ascending_order_it_should_pass()
```
Comment (dennisdoomen) [inline Tests/FluentAssertions.Specs/Collections/GenericCollectionAssertionsSpecs.cs:1290]: 🤔 Try to keep the names of test functional and never refer to the literal name of a method or API, e.g.

`When_the_collection_is_ordered_according_to_the_subsequent_ordering_assertion_it_should_succeed`

**Review by dennisdoomen — CHANGES_REQUESTED** [review]
👍 Nice work. 
❌ Please update the title to be functional

File: Tests/FluentAssertions.Specs/Collections/GenericCollectionAssertionsSpecs.cs [hunk 4]
```diff
@@ -1284,6 +1284,118 @@ public void When_asserting_the_items_in_ay_collection_are_not_ordered_using_an_i
...
+            collection.Should()
+                .BeInAscendingOrder(x => x.Item1)
+                .And
+                .ThenBeInAscendingOrder(x => x.Item2);
+        }
+
+        [Fact]
+        public void When_asserting_subsequence_by_be_in_ascending_order_it_should_fail()
+        {
+            var collection = new[]
```
Comment (jnyrup) [inline Tests/FluentAssertions.Specs/Collections/GenericCollectionAssertionsSpecs.cs:1309]: Please use the AAA-syntax.
```
// Arrange
// Act
// Assert
```

**Review by jnyrup — COMMENTED** [review]
Does this support multiple `ThenBeInAscendingOrder`?
E.g.
```cs
collection.Should()
	.BeInAscendingOrder(x => x.Item1)
	.And
	.ThenBeInAscendingOrder(x => x.Item2)
	.And
	.ThenBeInAscendingOrder(x => x.Item3);
```

**Comment by Serg046** [comment]
Applied the notes and added two tests showing multiple `ThenBeInAscending(Descending)Order` support

**Review by jnyrup — APPROVED** [review]

**Review by dennisdoomen — APPROVED** [review]

**Pull request merged**